shell/server/drivers/ip.js

```diff
@@ -234,16 +235,17 @@ const addressType = (address) => {
 };
 
 class IpNetworkImpl extends PersistentImpl {
-  constructor(db, saveTemplate) {
+  constructor(db, saveTemplate, tls) {
     super(db, saveTemplate);
+    this.tls = tls;
   }
 
   getRemoteHost(address) {
-    return { host: new IpRemoteHostImpl(address) };
+    return { host: new IpRemoteHostImpl(address, this.tls) };
```

Comment:
At first I thought "TLS authenticates hostnames, therefore the by-address method should probably throw an exception if using TLS", but apparently TLS can in theory authenticate IP addresses, it's just that none of the trusted CAs will ever sign such a certificate.

Since at present we don't provide a way to specify a custom CA, this path will never work for TLS, but it makes sense to keep here for later when we do provide custom CAs.

Maybe worth adding a comment?